packages/api/src/routes/medical/document.ts

```diff
@@ -108,29 +119,117 @@ router.post(
   })
 );
 
-/** ---------------------------------------------------------------------------
- * GET /downloadUrl
+/**
+ * Handles the logic for download url endpoints.
  *
+ * @param req Request object.
+ * @returns URL for downloading the document.
+ */
+async function getDownloadUrl(req: Request): Promise<string> {
+  const cxId = getCxIdOrFail(req);
+
+  const fileName = getFromQueryOrFail("fileName", req);
+  const type = getFrom("query").optional("conversionType", req);
+  const conversionType = type ? docConversionTypeSchema.parse(type) : undefined;
+
+  if ((typeof fileName !== "string" || fileName.indexOf(cxId) !== -1) && !Config.isSandbox()) {
+    throw new ForbiddenError();
+  }
+
+  const url = await downloadDocument({ fileName, conversionType });
+  return url;
+}
+
+// TODO: Redirect this endpoint to the new one (download-url)
+/** ---------------------------------------------------------------------------
+ * GET /document/downloadUrl
  * Fetches the document from S3 and sends a presigned URL
+ * @deprecated Use the GET /download-url endpoint instead.
  *
  * @param req.query.fileName The file name of the document in s3.
  * @param req.query.conversionType The doc type to convert to.
  * @return presigned url
  */
 router.get(
   "/downloadUrl",
+  asyncHandler(async (req: Request, res: Response) => {
+    const url = getDownloadUrl(req);
+    return res.status(MOVED_PERMANENTLY).json({ url });
+  })
+);
+
+/** ---------------------------------------------------------------------------
+ * GET /document/download-url
+ *
+ * Fetches the document from S3 and sends a presigned URL
+ *
+ * @param req.query.fileName The file name of the document in s3.
+ * @param req.query.conversionType The doc type to convert to.
+ * @return presigned url
+ */
+router.get(
+  "/download-url",
+  asyncHandler(async (req: Request, res: Response) => {
+    const url = getDownloadUrl(req);
+    return res.status(OK).json({ url });
+  })
+);
+
+/**
+ * POST /document/upload-url
+ *
+ * Uploads a medical document and creates a Document Reference for that file on the FHIR server.
+ *
+ * @param patientId - The ID of the patient.
+ * @body - The Partial<DocumentReference> draft to specify context for the file to be uploaded.
+ *
+ * @return documentReferenceID for the uploaded file.
```

Comment:
This needs updating. And I'd avoid talking about "draft" with customers... they just need to know the DocRef will be available once the document is uploaded. Less concepts make it simpler.